packages/squiggle-lang/__tests__/Symbolic_test.res

```diff
@@ -0,0 +1,30 @@
+open Jest
+open Expect
+open Js.Array
+open SymbolicDist
+
+let makeTest = (~only=false, str, item1, item2) =>
+  only
+    ? Only.test(str, () => expect(item1) -> toEqual(item2))
+    : test(str, () => expect(item1) -> toEqual(item2))
+
+let normalParams1: SymbolicDistTypes.normal = {mean: 5.0, stdev: 2.0}
+// let normalParams2: SymbolicDistTypes.normal = {mean: 10.0, stdev: 2.0}
+let normalParams3: SymbolicDistTypes.normal = {mean: 20.0, stdev: 2.0}
+let range20 = [0.0,1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0,10.0,11.0,12.0,13.0,14.0,15.0,16.0,17.0,18.0,19.0,20.0]
+let forSparkline = (thisPdf, inps) => map(thisPdf, inps)
+
+describe("Normal with Sparklines", () => {
+  let pdf1 = x => Normal.pdf(x, normalParams1)
+  let forSparkline1 = forSparkline(pdf1, range20)
+  makeTest("mean=5", Sparklines.sparkly(forSparkline1, ~options={minimum: None, maximum: None}), `▁▂▃▅███▅▃▂▁▁▁▁▁▁▁▁▁▁▁`)
+
+  let normal4 = Normal.add(normalParams1, normalParams3)
+  let normalParams4 = switch normal4 {
+      | #Normal(params) => params
+      | _ => {mean: 0.0, stdev: 1.0}
+  }
+  let pdf4 = x => Normal.pdf(x, normalParams4)
+  let forSparkline4 = forSparkline(pdf4, range20)
+  makeTest("mixture of two normals", Sparklines.sparkly(forSparkline4, ~options={minimum: None, maximum: None}), `▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▂▅█`)
```

Comment:
I feel like in general this testing code could be cleaned up a fair bit. I suggest taking another stab at changing up some things; then maybe I'll have a go

For example:
- I don't see where the "1" and "4" come from
- I don't get why the test-specific code isn't in blocks for the tests
- Some names aren't very descriptive; they describe where things are going, instead of what they are; i.e. "forSparkline".
- I feel like a lot of this could be abstracted into helper functions, so that the core test is very straightforward to read.

- Also, I'm not sure why the distribution at the end looks so weird, for the example. 
- Last, I don't think the word "mixture" is correct; a "mixture" is basically a pointwise addition. You could say "combination", or "Addition" to be more precise.

Reply:
yeah I'm recalling now re mixture--- a "mixture of two normals" refers to `p * N1 + (1 - p) * N2`, **very** different concept.

Reply:
[x] I don't get why the test-specific code isn't in blocks for the tests

Reply:
> Also, I'm not sure why the distribution at the end looks so weird, for the example. I think it's because it only goes to 20, but the distribution starts around 20.

that's because the underlying domain (`range20Float`) is fixed.

Reply:
> I don't see where the "1" and "4" come from

This is covered in another comment thread, upcoming commit.

Reply:
> instead of what they are; i.e. "forSparkline".

This function is now called `pdfImage`

Reply:
> I feel like a lot of this could be abstracted into helper functions, so that the core test is very straightforward to read.

I did this for parameter-wise addition.